app/src/main/java/org/dslul/openboard/inputmethod/latin/common/Colors.kt

```diff
@@ -24,39 +31,325 @@ import org.dslul.openboard.inputmethod.latin.KeyboardWrapperView
+
+        if (hasKeyBorders) {
+            backgroundStateList =
+                if (!isNight) stateList(adjustedFunctionalKey, background)
+                else stateList(adjustedKeyBackground, background)
+
+            keyStateList =
+                if (!isNight) stateList(adjustedBackground, keyBackground)
+                else stateList(adjustedKeyBackground, keyBackground)
+
+            functionalKeyStateList =
+                if (!isNight) stateList(doubleAdjustedFunctionalKey, functionalKey)
+                else stateList(functionalKey, doubleAdjustedKeyBackground)
+
+            actionKeyStateList =
+                if (!isNight) stateList(gesture, accent)
+                else stateList(doubleAdjustedAccent, accent)
+
+            spaceBarStateList =
+                if (themeStyle == STYLE_HOLO) stateList(spaceBar, spaceBar)
+                else keyStateList
+
+        } else {
+            // need to set color to background if key borders are disabled, or there will be ugly keys
+            backgroundStateList =
+                if (!isNight) stateList(adjustedFunctionalKey, background)
+                else stateList(adjustedKeyBackground, background)
+
+            keyStateList =
+                if (!isNight) stateList(adjustedFunctionalKey, Color.TRANSPARENT)
+                else stateList(functionalKey, Color.TRANSPARENT)
+
+            functionalKeyStateList =
+                if (themeStyle == STYLE_HOLO) stateList(functionalKey, Color.TRANSPARENT)
+                else keyStateList
+
+            actionKeyStateList =
+                if (themeStyle == STYLE_HOLO) stateList(accent, Color.TRANSPARENT)
+                else if (!isNight) stateList(gesture, accent)
+                else stateList(doubleAdjustedAccent, accent)
+
+            spaceBarStateList =
+                if (!isNight) stateList(gesture, adjustedFunctionalKey)
+                else stateList(adjustedKeyBackground, spaceBar)
+        }
+        keyTextFilter = colorFilter(keyText)
+
+        actionKeyIconColorFilter = when {
+            themeStyle == STYLE_HOLO -> keyTextFilter
+            // the white icon may not have enough contrast, and can't be adjusted by the user
+            isBrightColor(accent) -> colorFilter(Color.DKGRAY)
+            else -> null
+        }
+    }
+
+    override fun get(color: ColorType): Int = when (color) {
+        ColorType.ENABLE_TOOL_KEY, ColorType.EMOJI_CATEGORY_SELECTED -> accent
+        ColorType.EMOJI_CATEGORY, ColorType.GESTURE_PREVIEW -> adjustedBackground
+        ColorType.TOOL_BAR_KEY_BACKGROUND -> if (!isNight) accent else doubleAdjustedBackground
+        ColorType.GESTURE -> gesture
+        ColorType.KEY_TEXT, ColorType.COLOR_AUTO_CORRECT -> keyText
+        ColorType.SPACEBAR_TEXT -> spaceBarText
+        ColorType.NAV_BAR -> navBar
+        ColorType.MORE_SUGGESTIONS_HINT, ColorType.SUGGESTED_WORD, ColorType.TYPED_WORD, ColorType.VALID_WORD -> adjustedKeyText
+        else -> Color.WHITE
```

Comment:
_all_ `ColorType`s should return a chosen color. This is essentially a / the main reason to use enum and not some `int` assigned to each color.
When explicitly assigning a color to all values, it's impossible to forget setting a color (especially considering that new values might get added).

Reply:
Okay, I didn't know I had to list everything.
Done with commit a5af535.